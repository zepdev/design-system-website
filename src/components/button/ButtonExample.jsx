import React, { useState } from "react"
import { makeStyles } from "@material-ui/core/styles"
import Button from "@zlab-de/zel-react/Button"

const useStyles = makeStyles(theme => ({
  container: {
    display: "flex",
    justifyContent: "space-around",
    alignItems: "flex-end",
    padding: `${theme.space.s.rem}rem`,
    marginBottom: `${theme.space.s.rem}rem`,
  },
  heading: {
    marginBottom: `${theme.space.s.rem}rem`,
  },
  fullWidth: {
    border: `1px solid ${theme.color.global.lightGray}`,
    padding: `${theme.space.s.rem}rem`,
    width: 200,
  },
  text: {
    marginBottom: 30,
    textAlign: "center",
    fontSize: 18,
    fontWeight: 700,
  },
}))

function ButtonExample() {
  const classes = useStyles()
  const [text, setText] = useState("")

  return (
    <>
      <p data-testId="buttonText" className={classes.text}>
        You Clicked: {text}
      </p>
      <p className={classes.heading}>Primary</p>
      <div className={classes.container}>
        <Button variant="primary" onClick={() => setText("primary")}>
          Primary
        </Button>
        <Button
          variant="primary"
          size="medium"
          onClick={() => setText("primary medium")}
        >
          Primary Medium
        </Button>
        <Button
          variant="primary"
          size="small"
          onClick={() => setText("primary small")}
        >
          Primary Small
        </Button>
        <Button variant="primary" disabled>
          Primary Disabled
        </Button>
      </div>
      <div className={classes.container}>
        <Button
          variant="primary"
          fullWidth
          onClick={() => setText("primary full width")}
        >
          Primary Full Width
        </Button>
      </div>
      <p className={classes.heading}>Secondary</p>

      <div className={classes.container}>
        <Button variant="secondary" onClick={() => setText("secondary")}>
          Secondary
        </Button>
        <Button
          variant="secondary"
          size="medium"
          onClick={() => setText("secondary medium")}
        >
          Secondary Medium
        </Button>
        <Button
          variant="secondary"
          size="small"
          onClick={() => setText("secondary small")}
        >
          Secondary Small
        </Button>
        <Button variant="secondary" disabled>
          Secondary Disabled
        </Button>
      </div>
      <div className={classes.container}>
        <Button
          variant="secondary"
          fullWidth
          onClick={() => setText("secondary full width")}
        >
          Secondary Full Width
        </Button>
      </div>
    </>
  )
}

export default ButtonExample
