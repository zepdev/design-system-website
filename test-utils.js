import React from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "@material-ui/core/styles"
import theme from "./src/gatsby-theme-material-ui-top-layout/theme"

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
