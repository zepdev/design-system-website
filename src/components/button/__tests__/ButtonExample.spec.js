import React from "react"
import { render, fireEvent, screen } from "../../../../test-utils"
import "@testing-library/jest-dom/extend-expect"
import ButtonExample from "../ButtonExample"

describe("ButtonExample", () => {
  it("renders correctly", () => {
    const component = render(<ButtonExample />)
    expect(component).toMatchSnapshot()
  })
  it("renders button1", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: "Primary" }))
    expect(getByTestId("buttonText")).toHaveTextContent("You Clicked: primary")
  })
  it("renders button2", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: /Primary Medium/i }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: primary medium"
    )
  })
  it("renders button3", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: /Primary Small/i }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: primary small"
    )
  })
  it("renders button4", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: /Primary Full Width/i }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: primary full width"
    )
  })
  it("renders button5", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: "Secondary" }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: secondary"
    )
  })
  it("renders button6", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: "Secondary Medium" }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: secondary medium"
    )
  })
  it("renders button7", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(screen.getByRole("button", { name: "Secondary Small" }))
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: secondary small"
    )
  })
  it("renders button8", () => {
    const { getByTestId } = render(<ButtonExample />)
    fireEvent.click(
      screen.getByRole("button", { name: /secondary Full Width/i })
    )
    expect(getByTestId("buttonText")).toHaveTextContent(
      "You Clicked: secondary full width"
    )
  })
})
