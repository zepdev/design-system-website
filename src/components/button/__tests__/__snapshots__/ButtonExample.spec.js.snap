// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="makeStyles-text-4"
        data-testid="buttonText"
      >
        You Clicked: 
        
      </p>
      <p
        class="makeStyles-heading-2"
      >
        Primary
      </p>
      <div
        class="makeStyles-container-1"
      >
        <button
          class="zep-button zep-button--primary"
        >
          Primary
        </button>
        <button
          class="zep-button zep-button--primary zep-button--medium"
        >
          Primary Medium
        </button>
        <button
          class="zep-button zep-button--primary zep-button--small"
        >
          Primary Small
        </button>
        <button
          class="zep-button zep-button--primary"
          disabled=""
        >
          Primary Disabled
        </button>
      </div>
      <div
        class="makeStyles-container-1"
      >
        <button
          class="zep-button zep-button--full zep-button--primary"
        >
          Primary Full Width
        </button>
      </div>
      <p
        class="makeStyles-heading-2"
      >
        Secondary
      </p>
      <div
        class="makeStyles-container-1"
      >
        <button
          class="zep-button zep-button--secondary"
        >
          Secondary
        </button>
        <button
          class="zep-button zep-button--secondary zep-button--medium"
        >
          Secondary Medium
        </button>
        <button
          class="zep-button zep-button--secondary zep-button--small"
        >
          Secondary Small
        </button>
        <button
          class="zep-button zep-button--secondary"
          disabled=""
        >
          Secondary Disabled
        </button>
      </div>
      <div
        class="makeStyles-container-1"
      >
        <button
          class="zep-button zep-button--full zep-button--secondary"
        >
          Secondary Full Width
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="makeStyles-text-4"
      data-testid="buttonText"
    >
      You Clicked: 
      
    </p>
    <p
      class="makeStyles-heading-2"
    >
      Primary
    </p>
    <div
      class="makeStyles-container-1"
    >
      <button
        class="zep-button zep-button--primary"
      >
        Primary
      </button>
      <button
        class="zep-button zep-button--primary zep-button--medium"
      >
        Primary Medium
      </button>
      <button
        class="zep-button zep-button--primary zep-button--small"
      >
        Primary Small
      </button>
      <button
        class="zep-button zep-button--primary"
        disabled=""
      >
        Primary Disabled
      </button>
    </div>
    <div
      class="makeStyles-container-1"
    >
      <button
        class="zep-button zep-button--full zep-button--primary"
      >
        Primary Full Width
      </button>
    </div>
    <p
      class="makeStyles-heading-2"
    >
      Secondary
    </p>
    <div
      class="makeStyles-container-1"
    >
      <button
        class="zep-button zep-button--secondary"
      >
        Secondary
      </button>
      <button
        class="zep-button zep-button--secondary zep-button--medium"
      >
        Secondary Medium
      </button>
      <button
        class="zep-button zep-button--secondary zep-button--small"
      >
        Secondary Small
      </button>
      <button
        class="zep-button zep-button--secondary"
        disabled=""
      >
        Secondary Disabled
      </button>
    </div>
    <div
      class="makeStyles-container-1"
    >
      <button
        class="zep-button zep-button--full zep-button--secondary"
      >
        Secondary Full Width
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
